react_on_rails_pro/spec/dummy/client/app/components/CacheSection.jsx

```
@@ -0,0 +1,45 @@
+
+// CacheSection - Experimental component for section-level caching
+// Server component (no "use client" directive)
+// Each section waits for a delay before rendering (controlled by rake task)
+
+import React, { Suspense } from 'react';
+
+// Async wrapper that waits for the specified delay before rendering children
+async function AsyncSectionContent({ delayMs, children }) {
+  // Wait for the specified delay
+  if (delayMs > 0) {
+    await new Promise((resolve) => setTimeout(resolve, delayMs));
```

ESLint

[error] Return values from promise executor functions cannot be read.

(no-promise-executor-return)